File: backend2/CarShopBackEnd/CarShopBackEnd/Car.cs
using Microsoft.EntityFrameworkCore;
using System.Text.Json;

namespace CarShopBackEnd
{
    public class Car
    {
        public int id { get; set; }
        public string make { get; set; }
        public string model { get; set; }
        public string color { get; set; }
        public int volume { get; set; }
        public int mileage { get; set; }
        public int year { get; set; }
    }
}

File: car-shop-app/src/app/car.ts
export class Car {
    
    Id!: number;
    Make!: string; 
    Model!: string; 
    Color!: string;
    Volume!: number;
    Mileage!: number;
    Year!: number;

    static labels: { [id: string] : string; } = {
        Make: "Марка", 
        Model: "Модель", 
        Color: "Цвет", 
        Volume: "Объем, см3", 
        Mileage: "Пробег, км", 
        Year: "Год выпуска", 
    }

    static createFromValues(obj: any){
        var car = new Car();

        try {
            car.Id = obj.Id;
            car.Make = obj.Make;
            car.Model = obj.Model; 
            car.Color = obj.Color;
            car.Volume = obj.Volume;
            car.Mileage = obj.Mileage;
            car.Year = obj.Year;
            
            return car;
        }
        catch {
            return undefined;
        }
    }
}

